From 661a4dfb6de20f28dc337bf221608799f1b18a6e Mon Sep 17 00:00:00 2001
From: Graham Leggett <minfrin@apache.org>
Date: Sun, 26 Sep 2021 14:35:55 +0000
Subject: [PATCH] Backport:

*) mpm event: Correctly count active child processes in parent process if
   child process dies due to MaxConnectionsPerChild. PR 65592.
   Trunk version of patch:
      https://svn.apache.org/r1893520
   Backport version for 2.4.x of patch:
    Trunk version of patch works
    svn merge -c 1893520 ^/httpd/httpd/trunk .
   +1: rpluem, ylavic, minfrin



git-svn-id: https://svn.apache.org/repos/asf/httpd/httpd/branches/2.4.x@1893660 13f79535-47bb-0310-9956-ffa450edef68
---
 CHANGES                  | 4 ++++
 STATUS                   | 8 --------
 server/mpm/event/event.c | 5 ++++-
 3 files changed, 8 insertions(+), 9 deletions(-)

--- a/server/mpm/event/event.c
+++ b/server/mpm/event/event.c
@@ -2809,6 +2809,10 @@
         }
         ps = &ap_scoreboard_image->parent[i];
         if (ps->pid != 0) {
+            if (ps->quiescing == 1) {
+                ps->quiescing = 2;
+                active_daemons--;
+            }
             for (j = 0; j < threads_per_child; j++) {
                 ws = &ap_scoreboard_image->servers[i][j];
                 status = ws->status;
@@ -2888,7 +2892,6 @@
             ap_mpm_podx_signal(all_buckets[child_bucket].pod,
                                AP_MPM_PODX_GRACEFUL);
             retained->idle_spawn_rate[child_bucket] = 1;
-            active_daemons--;
         } else {
             ap_log_error(APLOG_MARK, APLOG_TRACE5, 0, ap_server_conf,
                          "Not shutting down child: total daemons %d / "
